File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//品牌服务层

app.service('brandService',function ($http) {
    this.findAll=function(){
        return  $http.get('../brand/findAll.do');
    };



    this.findPage = function (page, rows) {
        return $http.get('../brand/findPage.do?page=' + page + '&rows=' + rows);
    };
    this.save=function(id,entity) {
        var methodName = 'add';
        if (id != null) {//如果有ID
            methodName = 'update';//则执行修改方法
        }

        return  $http.post('../brand/' + methodName + '.do',entity);
    };


    this.findOne = function(id) {
        return $http.post('../brand/findOne.do?id=' + id);
    };



    this.dele= function(ids) {
        return $http.post('../brand/delete.do?ids=' + ids);
    };

    this.search=function(page,rows,searchEntity) {
        return $http.post('../brand/search.do?page=' + page + "&rows=" + rows,searchEntity);
    };

    //下拉列表数据
    this.selectOptionList=function(){
        return $http.get('../brand/selectOptionList.do');
    }

});

File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/service/BrandService.java
package com.pinyougou.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
    public List<TbBrand> findAll();
    /**
     * 返回分页列表
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);
    /**
     * 增加
     */
    public void add(TbBrand brand);
    /**
     * 删除
     * */
    public void delete(Long [] ids);

    /**
     * 修改
     * */
    public TbBrand findOne(Long id);
    public void update(TbBrand brand);

    /**
     * 分页
     * @param pageNum 当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(TbBrand brand, int pageNum,int pageSize);


    /**
     * 品牌下拉框数据
     */
    List<Map>selectOptionList();
}
